README.md
# Definition of Accounting
Accounting is the recording of financial transactions along with storing, sorting, retrieving, summarizing, and presenting the results in various reports and analyses. Accounting is also a field of study and profession dedicated to carrying out those tasks.

# Financial Accounting
One part of accounting focuses on presenting the financial information in the form of general-purpose financial statements (balance sheet, income statement, etc.) that are distributed to people outside of the company. These external reports must be prepared in accordance with generally accepted accounting principles often referred to as GAAP or US GAAP.

# Management Accounting
Another part of accounting focuses on providing a company's management with the information needed to keep the business financially healthy. Although some of the information comes from recorded transactions, many of the analyses and reports include estimated and projected amounts based on various assumptions. Generally, this information is not distributed to people outside of the company's management. A few examples of this information are budgets, standards for controlling operations, and estimating selling prices when quoting prices for new work.

# Other Examples of Accounting
Some of the many other examples of accounting include:

1. Advising on accounting systems
2. Income tax planning, advising, and reporting
3. Auditing the financial statements of companies and other organizations
4. Providing general business advice
5. Financial planning for individuals

# What is Py-Accounting?
Py-Accounting is work in progress that aims to help in creating the necessary documents used for analyzing and understanding the current business or personal financial status. Py-accounting hopes to be a useful if not very useful, personal, of small-scale help for anyone who may be having a hard time in doing productive things. 

# Why Python?
Python is an object-oriented, open-source, adaptable and simple to learn programming language. It has a rich arrangement of libraries and tools that makes the assignments simple for Data scientists.

Additionally, Python has an enormous community base where engineers and data scientists can solicit their queries and answer questions from others. Data science as services have been utilizing Python for quite a while and it will keep on being the top choice for Data scientists and Developers.

And since it so good with data manipulation, it is the go to for my little project.

# What is the end-product of Py-Accounting?
Since Python can do anything these days, the target end-product will be a word document completely done by Python, with pre-applied styling and proper calculations. 

Via <a href="https://python-docx.readthedocs.io/en/latest/">Python - Docx Project</a>

ChangeLog.md

# Python_In_Action
This is repository whose goal is to record equity and earning income. 
Currently a ice breaker for times when nothing is happening.

##Change Log
#### Date: 05/06/2022
1. Cleaner and better code for reading.
2. Made specific variables global such as
  
`global assets list`
`global Current_A_list`
`Current_A_list=[]`
`global Current_A_Value_List`
`Current_A_Value_List=[]`
`global Fixed_A_list`
`Fixed_A_list=[]`
`global Fixed_A_Value_List`
`Fixed_A_Value_List=[]`
`global intangible_A_list`
`intangible_A_list=[]`
`global intangible_A_Value_List`
`intangible_A_Value_List=[]`
`global Current_L_list`
`Current_L_list=[]`
`global Current_L_Value_List`
`Current_L_Value_List=[]`
`global Current_C_list`
`Current_C_list=[]`
`global Current_C_Value_List`
`Current_C_Value_List=[]`
`global user_choice`

3. Splitted the code of the intangible, fixed, and current assets so that I can work on them independently. 
4. Finally prints all of the assets.

#### 05/13/2022
1. Added more functionality.
2. Now checks whether the values you have entered follows the accounting equation.
3. New lists and variables namely `revenue_list`,`revenue_value_list`,`expense_list`,`expense_value_list`,`withdrawal`, and `withdrawal_value`.
4. Renamed `user_choices()` to `start_program()`
5. Copied and pasted the codes for expense, revenue, and withdrawals:
![image](https://user-images.githubusercontent.com/99703611/168235222-52cf0412-92a2-4214-a0cd-2bea9615aa81.png)

6. If the credit and debit side are not equal, it will notify the user and will also display the inequality.
7. Cleaned Up the Code

#### 06/09/2022
1. Now under new management. Will try to take down the original file because of change of principal author.

#### 06/10/2022
1. Changed a few more grammar-related questions.
2. Tweaked few choices for a better user experience.

#### 06/11/2022
1. Renamed "py-accounting.py.py" to "main.py"
2. Changed most variable names to easier identify what entry to use.
![IMG_20220611_145831](https://user-images.githubusercontent.com/107169094/173177280-da0e0709-8e9e-46e0-86cb-797ed3070553.jpg)

#### 07/04/2022
1. Quality of Life update
    Previouslywhen recording, all string values for records are all inputted first before their corresponding value. Now, the recording is now done in intervals where the string value is recorded first, then the value, then the loop repeats untilthe specified number is met.


#### Known Issues
1. Fixing a bug that does not allow SL4a to run the script on older android tablets.
2. Still no .exe file version.

Do note that this is a long-term project. This is still in phase 1. 
### More things Will be added


Main.py
#phase 1 of the project

import time
from docx import *
from docx.enum.style import *
from docx.shared import *
from docx.oxml.ns import *
from docx.enum.section import WD_SECTION
from docx.enum.text import *
time.sleep(0.5)

time.sleep(1)
def variables():
#global assets list
    global Current_Assets
    Current_Assets=[]
    global Current_Assets_Value
    Current_Assets_Value=[]
    global Fixed_Assets
    Fixed_Assets=[]
    global Fixed_Assets_Value
    Fixed_Assets_Value=[]
    global Intangible_Assets
    Intangible_Assets=[]
    global Intangible_Assets_Value
    Intangible_Assets_Value=[]
    #global liability list
    global Current_Liabilities
    Current_Liabilities=[]
    global Current_Liabilities_Amount
    Current_Liabilities_Amount=[]
    #global Capital_Value (or equity depending on your region) List
    global Capital_Entry
    Capital_Entry=[]
    global Capital_Value
    Capital_Value=[]
    #extended global variables
    global Revenues
    Revenues=[]
    global Revenue_Quantity
    Revenue_Quantity=[]
    global Expenditures
    Expenditures=[]
    global Expenditure_Cost
    Expenditure_Cost=[]
    global Withdrawal
    Withdrawal=[]
    global Withdrawal_value
    Withdrawal_value=[]
    #global variables
    global user_choice

def start_program(): #The very first thing that happens in the process. 
    print('What do you choose to Record?')
    time.sleep(1.5)
    user_choice = input("Enter 'Assets', 'Liabilities', or 'Capital':")
    print('-'*20)
    time.sleep(0.5)

    if user_choice == 'Assets': #If the user entered 'Assets' in the first question.
        asset_entry = input("What entry of assets is it? 'F' for fixed, 'C' for current, 'I' for intangible:")
        if asset_entry == 'C':
            current_asset_record()
        elif asset_entry == 'F':
            fixed_asset_record()
        elif asset_entry =='I':
            intangible_asset_record()
        else: #If neither of the user's choice was 'C', 'F', or 'I'.
            print('Sorry, that is an invalid syntax.')
            time.sleep(0.5)
            start_program()
    elif user_choice == 'Capital': #if the user entered 'Capital' in the first question.
        capital_entry = input("'R' for revenue and 'E' for expenses. If none, enter [n]:")
        if capital_entry == 'R':
            revenue_record()
        elif capital_entry == 'E':
            expense_record()
        elif capital_entry == 'n':
            recording(user_choice)
        else:
            print('That is an invalid syntax. Try Again.')
            start_program()
    else:
        recording(user_choice)

#asks for specific information regarding the user_choice part
def choice_handle(user_choice):
	recording(user_choice)

def Withdrawal_record(): #Recording of withdrawals
    n = int(input("How many Withdrawal entries incurred?: "))

    print("\n")
    for i in range(0, n):
        print("Withdrawal #", i+1, ':', sep='')
        withdraw = input()
        Withdrawal.append(withdraw)
        print("You have entered ", Withdrawal)
        print("Value of Withdrawal #", i+1, ':', sep='')
        withdrawn = int(input())
        Withdrawal_value.append(withdrawn)
    #shows what user have entered
    print('You have entered the following account title(s):',Withdrawal)
    print('You have entered the following asset value:',Withdrawal_value)
    time.sleep(1)
    #user confirmation
    user_confirmation=input('is the following information correct?:')
    if user_confirmation=='y':
        Retry_parameter()
    elif user_confirmation=='n':
        time.sleep(0.5)
        recording(user_choice)    

def expense_record(): #recording of expenses
    n = int(input("How many Expense Entries incurred?: "))

    print("\n")
    for i in range(0, n):
        print("Expense #", i+1, ':', sep='')
        entry = input()
        Expenditures.append(entry)
        print("You have entered ", Expenditures)
        print("Value of Expense #", i+1, ':', sep='')
        value = int(input())
        Expenditure_Cost.append(value)
    #shows what user have entered
    print('You have entered the following account title(s):',Expenditures)
    print('You have entered the following asset value:',Expenditure_Cost)
    time.sleep(1)
    #user confirmation
    user_confirmation=input('is the following information correct?:')
    if user_confirmation=='y':
        Retry_parameter()
    elif user_confirmation=='n':
        time.sleep(0.5)
        recording(user_choice)   

def revenue_record(): #Recording of Revenue
    n = int(input("How many Revenue Entries incurred?: "))

    print("\n")
    for i in range(0, n):
        print("Revenue #", i+1, ':', sep='')
        entry = input()
        Revenues.append(entry)
        print("You have entered ", Revenues)
        print("Value of Revenue #", i+1, ':', sep='')
        value = int(input())
        Revenue_Quantity.append(value)
    #shows what user have entered
    print('You have entered the following account title(s):',Revenues)
    print('You have entered the following asset value:',Revenue_Quantity)
    time.sleep(1)
    #user confirmation
    user_confirmation=input('is the following information correct?:')
    if user_confirmation=='y':
        Retry_parameter()
    elif user_confirmation=='n':
        time.sleep(0.5)
        recording(user_choice)

def intangible_asset_record(): #Recording of Intangible Assets such as goodwill or customer care service ratings.

    n = int(input("How many Assets?: "))

    print("\n")
    for i in range(0, n):
        print("Asset #", i+1, ':', sep='')
        entry = input()
        Intangible_Assets.append(entry)
        print("You have entered ", Intangible_Assets)
        print("Value of Asset #", i+1, ':', sep='')
        value = int(input())
        Intangible_Assets_Value.append(value)
    #shows what user have entered
    print('You have entered the following account title(s):',Intangible_Assets)
    print('You have entered the following asset value:',Intangible_Assets_Value)
    time.sleep(1)
    #user confirmation
    user_confirmation=input('is the following information correct?:')
    if user_confirmation=='y':
        Retry_parameter()
    elif user_confirmation=='n':
        time.sleep(0.5)
        recording(user_choice)

def current_asset_record(): #Recording of current assets such as cash, furnitures, etc.
    n = int(input("How many assets?: "))

    print("\n")
    for i in range(0, n):
        print("Asset #", i+1, ':', sep='')
        entry = input()
        Current_Assets.append(entry)
        print("You have entered ", Current_Assets)
        print("Value of Asset #", i+1, ':', sep='')
        value = int(input())
        Current_Assets_Value.append(value)
 
    #shows what user have entered
    print('You have entered the following account title(s):',Current_Assets)
    print('You have entered the following asset value:',Current_Assets_Value)
    time.sleep(1)
    #user confirmation
    user_confirmation = input('is the following information correct?[y,n]:')
    if user_confirmation == 'y':
        Retry_parameter()
    elif user_confirmation == 'n':
        time.sleep(0.5)
        recording(user_choice)
        
def fixed_asset_record(): #Recording of fixed assets such as building and land.

    n = int(input("How many assets?: "))

    print("\n")
    for i in range(0, n):
        print("Asset #", i+1, ':', sep='')
        entry = input()
        Fixed_Assets.append(entry)
        print("You have entered ", Fixed_Assets)
        print("Value of Asset #", i+1, ':', sep='')
        value = int(input())
        Fixed_Assets_Value.append(value)
    #shows what user have entered
    print('You have entered the following account title(s):',Fixed_Assets)
    print('You have entered the following asset value:',Fixed_Assets_Value)
    time.sleep(1)
    #user confirmation
    user_confirmation=input('is the following information correct?:')
    if user_confirmation=='y':
        Retry_parameter()
    elif user_confirmation=='n':
        time.sleep(0.5)
        recording(user_choice)

def recording(user_choice):
    if user_choice=='Liabilities': #Recording of liability such as accounts and tax payables.
        n = int(input("How many liabilities?: "))

        print("\n")
        for i in range(0, n):
            print("Liability #", i+1, ':', sep='')
            entry = input()
            Current_Liabilities.append(entry)
            print("You have entered ", Current_Liabilities)
            print("Value of Liability #", i+1, ':', sep='')
            value = int(input())
            Current_Liabilities_Amount.append(value)

        #shows what user have entered
        print('You have entered the following account title(s):',Current_Liabilities)
        print('You have entered the following liability values(s)',Current_Liabilities_Amount)
        time.sleep(1)
        #user confirmation
        user_confirmation=input('is the following information correct?[y,n]:')
        if user_confirmation=='y':
            Retry_parameter()
        elif user_confirmation=='n':
            time.sleep(0.5)
            recording(user_choice)
    elif user_choice=='Capital': #Recording of capital/equity of the owner or a shareholder.
        n = int(input("How many assets?: "))

        print("\n")
        for i in range(0, n):
            print("Capital/Equity #", i+1, ':', sep='')
            entry = input()
            Capital_Entry.append(entry)
            print("You have entered ", Capital_Entry)
            print("Value of Capital/Equity #", i+1, ':', sep='')
            value = int(input())
            Capital_Value.append(value)
        #shows what user have entered
        print('You have entered the following account title(s):',Capital_Entry)
        print('You have entered the following capital values(s)',Capital_Value)
        time.sleep(0.5)
        #user confirmation
        user_confirmation=input('is the following information correct?[y,n]:')
        if user_confirmation=='y':
            Retry_parameter()
        elif user_confirmation=='n':
            time.sleep(0.5)
            recording(user_choice)
    else: #Catches all uneccessary inputs to avoid encountering errors.
        print("Only 'l' or 'c' . Try again")
        time.sleep(0.5)
        start_program()

def Retry_parameter(): #Only appears when the recorded values are correct and asks if the user want to record again.
    time.sleep(0.5)
    again=input('Would you like to record again? [y,n]: ')
    if again=='y':
        start_program()
    elif again=='n':
        print('The values you have recorded are now stored on RAM. To avoid loss, do not quit the application.')
        print('-'*20)
        for a, b in zip(Current_Assets, Current_Assets_Value):
            print(a, b)  
        print('-'*20)
        for a, b in zip(Fixed_Assets, Fixed_Assets_Value):
            print(a, b)
        print('-'*20)
        for a, b in zip(Intangible_Assets, Intangible_Assets_Value):
            print(a, b)
        print('-'*20)
        for a, b in zip(Current_Liabilities, Current_Liabilities_Amount):
            print(a, b)
        print('-'*20)
        for a, b in zip(Capital_Entry, Capital_Value):
            print(a, b)
        print('-'*20)
        for a, b in zip(Revenues, Revenue_Quantity):
            print(a,b)
        print('-'*20)
        for a, b in zip(Expenditures, Expenditure_Cost):
            print(a, b)
        print('-'*20)
        for a, b in zip(Withdrawal, Withdrawal_value):
            print(a, b)

#Phase 2 of The Project
def phase2():
    document = Document()
    sections = document.sections
    for section in sections:
        section.top_margin = Inches(0.5)
        section.bottom_margin = Inches(0.5)
        section.left_margin = Inches(1)
        section.right_margin = Inches(1)

    section = document.sections[0]

    sectPR =  section._sectPr
    cols = sectPR.xpath('./w:cols')[0]

    cols.set(qn('w:num'), '1')

    document.add_heading('Balance Sheet', 0)

    global content
    content = 'Hello, thank you for using Py-accounting software. You May Delete this Message.'

    def writedocx(content, font_name = 'Times New Roman', font_size = 12, font_bold = False, font_italic = False, font_underline = False, color = RGBColor(0, 0, 0),
                before_spacing = 5, after_spacing = 5, line_spacing = 1.5, keep_together = True, keep_with_next = False, page_break_before = False,
                widow_control = False, align = 'left', style = ''):
        paragraph = document.add_paragraph(str(content))
        paragraph.style = document.styles.add_style(style, WD_STYLE_TYPE.PARAGRAPH)
        font = paragraph.style.font
        font.name = font_name
        font.size = Pt(font_size)
        font.bold = font_bold
        font.italic = font_italic
        font.underline = font_underline
        font.color.rgb = color
        paragraph_format = paragraph.paragraph_format
        paragraph_format.space_before = Pt(before_spacing)
        paragraph_format.space_after = Pt(after_spacing)
        paragraph.line_spacing = line_spacing
        paragraph_format.keep_together = keep_together
        paragraph_format.keep_with_next = keep_with_next
        paragraph_format.page_break_before = page_break_before
        paragraph_format.widow_control = widow_control
        if align.lower() == 'left':
            paragraph.alignment = WD_PARAGRAPH_ALIGNMENT.LEFT
        elif align.lower() == 'center':
            paragraph.alignment = WD_PARAGRAPH_ALIGNMENT.CENTER
        elif align.lower() == 'right':
            paragraph.alignment = WD_PARAGRAPH_ALIGNMENT.RIGHT
        elif align.lower() == 'justify':
            paragraph.alignment = WD_PARAGRAPH_ALIGNMENT.JUSTIFY
        else:
            paragraph.alignment = WD_PARAGRAPH_ALIGNMENT.LEFT
    writedocx(content) 

    document.save('Word.docx')

variables()
start_program()

check=input('Would you like to check for imbalances in both credit side and debit side? [y,n]:')
if check=='y':
    if sum(Current_Assets_Value)+sum(Fixed_Assets_Value)+sum(Intangible_Assets_Value)==sum(Current_Liabilities_Amount)+sum(Capital_Value)+sum(Revenue_Quantity)-sum(Expenditure_Cost)-sum(Withdrawal_value):
        time.sleep(0.5)
        print('OK')
    else:
        time.sleep(0.5)
        print('The inputted values are not equal.')
        if sum(Current_Assets_Value)+sum(Fixed_Assets_Value)+sum(Intangible_Assets_Value)>sum(Current_Liabilities_Amount)+sum(Capital_Value)+sum(Revenue_Quantity)-sum(Expenditure_Cost)-sum(Withdrawal_value):
            print('The debit side is higher then the credit side.')
            time.sleep(1)
            print('The debit side is', sum(Current_Assets_Value)+sum(Fixed_Assets_Value)+sum(Intangible_Assets_Value),'and the credit side is', sum(Current_Liabilities_Amount)+sum(Capital_Value)+sum(Revenue_Quantity)-sum(Expenditure_Cost)-sum(Withdrawal_value))
            seeif=input('Would you like to fix the errors? [y,n]:')
            if seeif=='y':
                start_program()
            elif seeif=='n':
                print('ok.')
        if sum(Current_Assets_Value)+sum(Fixed_Assets_Value)+sum(Intangible_Assets_Value)<sum(Current_Liabilities_Amount)+sum(Capital_Value)+sum(Revenue_Quantity)-sum(Expenditure_Cost)-sum(Withdrawal_value):
            print('The credit side is higher than the debit side.')
            time.sleep(1)
            print('The debit side is', sum(Current_Assets_Value)+sum(Fixed_Assets_Value)+sum(Intangible_Assets_Value),'and the credit side is', sum(Current_Liabilities_Amount)+sum(Capital_Value)+sum(Revenue_Quantity)-sum(Expenditure_Cost)-sum(Withdrawal_value))
            seeif=input('Would you like to fix the errors? [y,n]:')
            if seeif=='y':
                start_program()
            elif seeif=='n':
                print('ok.')
else:
    phase2()




Document_creation.py
from docx import *
from docx.enum.style import *
from docx.shared import *
from docx.oxml.ns import *
from docx.enum.section import WD_SECTION
from docx.enum.text import *

document = Document()
sections = document.sections
for section in sections:
    section.top_margin = Inches(0.5)
    section.bottom_margin = Inches(0.5)
    section.left_margin = Inches(1)
    section.right_margin = Inches(1)

section = document.sections[0]

sectPR =  section._sectPr
cols = sectPR.xpath('./w:cols')[0]

cols.set(qn('w:num'), '1')

document.add_heading('Balance Sheet', 0)

global content
content = 'Hello, thank you for using Py-accounting software. You May Delete this Message.'

def writedocx(content, font_name = 'Times New Roman', font_size = 12, font_bold = False, font_italic = False, font_underline = False, color = RGBColor(0, 0, 0),
              before_spacing = 5, after_spacing = 5, line_spacing = 1.5, keep_together = True, keep_with_next = False, page_break_before = False,
              widow_control = False, align = 'left', style = ''):
    paragraph = document.add_paragraph(str(content))
    paragraph.style = document.styles.add_style(style, WD_STYLE_TYPE.PARAGRAPH)
    font = paragraph.style.font
    font.name = font_name
    font.size = Pt(font_size)
    font.bold = font_bold
    font.italic = font_italic
    font.underline = font_underline
    font.color.rgb = color
    paragraph_format = paragraph.paragraph_format
    paragraph_format.space_before = Pt(before_spacing)
    paragraph_format.space_after = Pt(after_spacing)
    paragraph.line_spacing = line_spacing
    paragraph_format.keep_together = keep_together
    paragraph_format.keep_with_next = keep_with_next
    paragraph_format.page_break_before = page_break_before
    paragraph_format.widow_control = widow_control
    if align.lower() == 'left':
        paragraph.alignment = WD_PARAGRAPH_ALIGNMENT.LEFT
    elif align.lower() == 'center':
        paragraph.alignment = WD_PARAGRAPH_ALIGNMENT.CENTER
    elif align.lower() == 'right':
        paragraph.alignment = WD_PARAGRAPH_ALIGNMENT.RIGHT
    elif align.lower() == 'justify':
        paragraph.alignment = WD_PARAGRAPH_ALIGNMENT.JUSTIFY
    else:
        paragraph.alignment = WD_PARAGRAPH_ALIGNMENT.LEFT
writedocx(content)

document.save('Word.docx')


Equity_Recorder.py
#pylint:disable=E0602
#A program that records user input labelling it as either Assets , Liabilities, or Equity.

import time
def global_things():
    time.sleep(0.5)
    #global assets list
    global Current_A_list
    Current_A_list=[]
    global Current_A_Value_List
    Current_A_Value_List=[]
    global Fixed_A_list
    Fixed_A_list=[]
    global Fixed_A_Value_List
    Fixed_A_Value_List=[]
    global intangible_A_list
    intangible_A_list=[]
    global intangible_A_Value_List
    intangible_A_Value_List=[]
    #global liability list
    global Current_L_list
    Current_L_list=[]
    global Current_L_Value_List
    Current_L_Value_List=[]
    #global Capital List
    global Current_C_list
    Current_C_list=[]
    global Current_C_Value_List
    Current_C_Value_List=[]
    #global variables
    global user_choice

def user_choices():
    print('What do you choose to Record?')
    time.sleep(1.5)
    user_choice=input('a for assets, l for liabilities, and c for Capital:')
    if user_choice=='a':
        asset_type=input("'F' for fixed, 'C' for current, 'I' for intangible")
        if asset_type=='C':
            current_asset_record()
        elif asset_type=='F':
            fixed_asset_record()
        elif asset_type=='I':
            intangible_asset_record()
        else:
            print('Sorry, that is an invalid syntax.')
            time.sleep(0.5)
            user_choices()
    else:
        recording(user_choice)

#asks for specific information regarding the user_choice part
def choice_handle(user_choice):
	recording(user_choice)

def intangible_asset_record():
    print('When recording, do not use spaces, rather, use underscores')
    time.sleep(0.5)
    print('Examples: notes payable should be notes_payable')
    time.sleep(0.5)
    print('As much as possible, capitalize the first letter.')
    time.sleep(0.5)
    print('When adding records, always first write on paper to avoid loss.')
    time.sleep(0.5)
    print('If you do not have a paper, you may also restart the program.')
    time.sleep(0.5)

    n = int(input("How many asset account titles?: "))

    print("\n")
    for i in range(0, n):
        print("Enter the asset", i, )
        asset = input()
        intangible_A_list.append(asset)
    print("You have entered ", intangible_A_list)
    time.sleep(0.5)
    print("\n")
    for i in range(0, n):
        print("Enter the asset value for ", i, )
        assetvalue = input()
        intangible_A_Value_List.append(assetvalue)
    print("You have entered ", intangible_A_Value_List)
    time.sleep(0.5)
    #shows what user have entered
    print('You have entered the following account title(s):',intangible_A_list)
    print('You have entered the following asset value:',intangible_A_Value_List)
    time.sleep(1)
    #user confirmation
    user_confirmation=input('is the following information correct?:')
    if user_confirmation=='y':
        Retry_parameter()
    elif user_confirmation=='n':
        time.sleep(0.5)
        recording(user_choice)
def current_asset_record():
    print('When recording, do not use spaces, rather, use underscores')
    time.sleep(0.5)
    print('Examples: notes payable should be notes_payable')
    time.sleep(0.5)
    print('As much as possible, capitalize the first letter.')
    time.sleep(0.5)
    print('When adding records, always first write on paper to avoid loss.')
    time.sleep(0.5)
    print('If you do not have a paper, you may also restart the program.')
    time.sleep(0.5)
    n = int(input("How many asset account titles?: "))
    print("\n")
    for i in range(0, n):
        print("Enter the asset", i, )
        asset = input()
        Current_A_list.append(asset)
    print("You have entered ", Current_A_list)
    time.sleep(0.5)
    print("\n")
    for i in range(0, n):
        print("Enter the asset value for ", i, )
        assetvalue = input()
        Current_A_Value_List.append(assetvalue)
    print("You have entered ", Current_A_Value_List)
    time.sleep(0.5)
    #shows what user have entered
    print('You have entered the following account title(s):',Current_A_list)
    print('You have entered the following asset value:',Current_A_Value_List)
    time.sleep(1)
    #user confirmation
    user_confirmation=input('is the following information correct?:')
    if user_confirmation=='y':
        Retry_parameter()
    elif user_confirmation=='n':
        time.sleep(0.5)
        recording(user_choice)
def fixed_asset_record():
    print('When recording, do not use spaces, rather, use underscores')
    time.sleep(0.5)
    print('Examples: notes payable should be notes_payable')
    time.sleep(0.5)
    print('As much as possible, capitalize the first letter.')
    time.sleep(0.5)
    print('When adding records, always first write on paper to avoid loss.')
    time.sleep(0.5)
    print('If you do not have a paper, you may also restart the program.')
    time.sleep(0.5)
    n = int(input("How many asset account titles?: "))

    print("\n")
    for i in range(0, n):
        print("Enter the asset", i, )
        asset = input()
        Fixed_A_list.append(asset)
    print("You have entered ", Fixed_A_list)
    time.sleep(0.5)
    print("\n")
    for i in range(0, n):
        print("Enter the asset value for ", i, )
        assetvalue = input()
        Fixed_A_Value_List.append(assetvalue)
    print("You have entered ", Fixed_A_Value_List)
    time.sleep(0.5)
    #shows what user have entered
    print('You have entered the following account title(s):',Fixed_A_list)
    print('You have entered the following asset value:',Fixed_A_Value_List)
    time.sleep(1)
    #user confirmation
    user_confirmation=input('is the following information correct?:')
    if user_confirmation=='y':
        Retry_parameter()
    elif user_confirmation=='n':
        time.sleep(0.5)
        recording(user_choice)

def recording(user_choice):
    if user_choice=='l':
        n = int(input("How many liability account titles?: "))

        print("\n")
        for i in range(0, n):
            print("Enter the liability", i, )
            liability = input()
            Current_L_list.append(liability)
        print("You have entered ", Current_L_list)
        time.sleep(0.5)
        print("\n")
        for i in range(0, n):
            print("Enter the asset value for ", i, )
            liabilityvalue = input()
            Current_L_Value_List.append(liabilityvalue)
        time.sleep(0.5)
        #shows what user have entered
        print('You have entered the following account title(s):',Current_L_list)
        print('You have entered the following liability values(s)',Current_L_Value_List)
        time.sleep(1)
        #user confirmation
        user_confirmation=input('is the following information correct?[y,n]:')
        if user_confirmation=='y':
            Retry_parameter()
        elif user_confirmation=='n':
            time.sleep(0.5)
            recording(user_choice)
    elif user_choice=='c':
        n = int(input("How many Capital account titles?: "))

        print("\n")
        for i in range(0, n):
            print("Enter the Capital Account Title", i, )
            capital = input()
            Current_C_list.append(capital)
        print("You have entered ", Current_C_list)
        time.sleep(0.5)
        print("\n")
        for i in range(0, n):
            print("Enter the Capital value for ", i, )
            capitalvalue = input()
            Current_C_Value_List.append(capitalvalue)
        time.sleep(0.5)
        #shows what user have entered
        print('You have entered the following account title(s):',Current_C_list)
        print('You have entered the following capital values(s)',Current_C_Value_List)
        time.sleep(0.5)
        #user confirmation
        user_confirmation=input('is the following information correct?[y,n]:')
        if user_confirmation=='y':
            Retry_parameter()
        elif user_confirmation=='n':
            time.sleep(0.5)
            recording(user_choice)
    else:
        print('ony a or b or c. Try again')
        time.sleep(0.5)
        user_choices()

def Retry_parameter():
    time.sleep(0.5)
    again=input('Would you like to record again? [y,n]: ')
    if again=='y':
        user_choices()
    elif again=='n':
        print('Your values have been saved')
        print('Current Assets:', Current_A_list)
        print('Value', Current_A_Value_List)
        print('Fixed Assets:', Fixed_A_list)
        print('Value', Fixed_A_Value_List)
        print('Intangible Assets:', Current_A_list)
        print('Value', intangible_A_Value_List)
        print('Liabilities:', Current_L_list)
        print('Value', Current_L_Value_List)
        print('Capital:', Current_C_list)
        print('Value', Current_C_Value_List)


global_things()
user_choices()   

	